src/router.lisp

```diff
@@ -29,7 +29,7 @@
              (new (if (< old (1- len))
                       (1+ old)
                       0)))
-        (atomic:atomic-cas index old new)
+        (atomic:atomic-setf index new)
```

Comment:
`atomic-setf` retrieves the current value of the atomic and sets the new. What happens if the old value is already beyond the new value?

Reply:
It's fixed and pushed master, but I was still failed to find how to rebase when a PR had already created, please help me, thanks.